### PR TITLE
PDI-13780 - Database Lookup yields wrong results when preloading table into cache

### PR DESCRIPTION
**Pull request by yuryilyukevich**
@mattyb149, @brosander, review it please.

### CODE REVIEW

**Comment by yuryilyukevich** [comment]
@mattyb149, @brosander, review it please.

File: engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java [hunk 1]
```diff
@@ -77,7 +76,7 @@ public DatabaseLookup( StepMeta stepMeta, StepDataInterface stepDataInterface, i
    * @return the resulting row after the lookup values where added
    * @throws KettleException In case something goes wrong.
    */
-  private synchronized Object[] lookupValues( RowMetaInterface inputRowMeta, Object[] row ) throws KettleException {
+  private Object[] lookupValues( RowMetaInterface inputRowMeta, Object[] row ) throws KettleException {
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java]: Why remove synchronized?

File: engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java [hunk 2]
```diff
@@ -215,7 +214,7 @@ public DatabaseLookup( StepMeta stepMeta, StepDataInterface stepDataInterface, i
     return outputRow;
   }
 
-  private void storeRowInCache( RowMetaInterface lookupMeta, Object[] lookupRow, Object[] add ) {
+  void storeRowInCache( RowMetaInterface lookupMeta, Object[] lookupRow, Object[] add ) {
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java]: If these are package-private for a reason, please add Javadoc and/or annotations. If it is to expose this method for testing, we've been using Guava's @VisibleForTesting annotation.

**Comment by yuryilyukevich** [comment]
@mattyb149, Commit was updated. Review it please.

**Pull request merged**